files/scripts/startup_tsa_tsb.py

```diff
@@ -69,19 +69,42 @@ def config_tsa():
     if tsa_ena == True:
         logger.log_info("Configuring TSA")
         subprocess.check_output(['TSA']).strip()
+        logger.log_info("Setting TSA-TSB service field in STATE_DB")
+        subprocess.check_output([
+            'sonic-db-cli', 'STATE_DB', 'HSET', 'ALL_SERVICE_STATUS|tsa_tsb_service', 'running', 'OK'
+        ]).strip()
     else:
-        if num_asics > 1:
-            logger.log_info("Either TSA is already configured or switch sub_role is not Frontend - not configuring TSA")
+        #check if tsa_tsb service is already running, restart the timer
+        try:
+            startup_tsa_tsb_service_status = subprocess.check_output([
+                'sonic-db-cli', 'STATE_DB', 'HGET', 'ALL_SERVICE_STATUS|tsa_tsb_service', 'running'
+            ]).strip().decode('utf-8')  # Convert bytes to string
+        except subprocess.CalledProcessError:
+            startup_tsa_tsb_service_status = None  # Default if the field is missing
+
+        if startup_tsa_tsb_service_status == 'OK':
+            logger.log_info("TSA-TSB service is already running, just restart the timer")
+            return True
         else:
-            logger.log_info("Either TSA is already configured - not configuring TSA")
+            if num_asics > 1:
+                logger.log_info("Either TSA is already configured or switch sub_role is not Frontend - not configuring TSA")
```

Comment:
Is `num_asics > 1` the right thing to check here given we will likely have multi-asic fixed systems in the future? Is there a check that is closer to `is_chassis` we can use instead?

Reply:
@arista-nwolfe yes, we could do a better check here. For this PR, I wanted to keep the code changes minimal, hence didnt modify this existing check. We can look to enhance it in future.